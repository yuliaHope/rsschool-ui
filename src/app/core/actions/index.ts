export * from './schedule';
export * from './user';
export * from './courses';
export * from './assignments';
