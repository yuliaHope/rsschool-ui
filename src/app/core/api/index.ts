export * from './schedule';
export * from './session';
export * from './user';
export * from './courses';
export * from './assignments';
