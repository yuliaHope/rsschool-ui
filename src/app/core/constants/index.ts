export * from './schedule';
export * from './courses';
export * from './user';
export * from './assignments';
