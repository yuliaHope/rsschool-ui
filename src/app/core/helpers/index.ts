export * from './schedule';
export * from './assignment';
