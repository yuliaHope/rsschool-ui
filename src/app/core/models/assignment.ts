export enum AssignmentStatus {
    Assigned = 'Assigned',
    ReadyForReview = 'ReadyForReview',
    Rejected = 'Rejected',
    Checked = 'Checked',
}

export interface IAssignment {
    taskId: number;
    courseId: string;
    studentId: string;
    mentorId: string;
    studentComment: string;
    mentorComment: string;
    score: number;
    assignmentRepo: string;
    deadlineDate: number;
    completeDate: number;
    checkDate: number;
    status: AssignmentStatus;
    [key: number]: any;
}

export interface IAssignmentDocument extends IAssignment {
    _id: string;
}
