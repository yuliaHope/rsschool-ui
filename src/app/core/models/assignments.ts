export enum AssignmentStatus {
    Assigned = 'Assigned',
    ReadyForReview = 'ReadyForReview',
    Rejected = 'Rejected',
    Checked = 'Checked',
}

export interface IAssignment {
    assignmentRepo: string;
    checkDate?: number;
    completeDate?: number;
    courseId: string;
    deadlineDate: number;
    mentorComment: string;
    mentorId: string;
    score: number;
    status: AssignmentStatus;
    studentComment: string;
    studentId: string;
    taskId: string;
    title: string;
    urlToDescription?: string;
    whoChecks?: string;
}

export interface IAssignmentDocument extends IAssignment {
    _id: number;
}
