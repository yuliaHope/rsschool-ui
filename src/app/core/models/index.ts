export * from './events';
export * from './stage';
export * from './user';
export * from './course';
export * from './courseMentor';
export * from './courseStudent';
export * from './assignment';
