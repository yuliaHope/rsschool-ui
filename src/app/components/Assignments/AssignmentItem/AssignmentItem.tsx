import * as React from 'react';
import { Card, CardHeader } from 'reactstrap';
import AssignmentItemBody from '../AssignmentItemBody/AssignmentItemBody';
import AssignmentItemForm, { AssignmentFormData } from '../AssignmentItemForm/AssignmentItemForm';
import { INormalizeAssignment } from 'core/reducers/assignments';
import { classNames } from 'core/styles';
import { setCardStyle } from 'core/helpers';

const cn = classNames(require('../index.scss'));

type AssignmentItemProps = {
    assignment: INormalizeAssignment;
    updateAssignment: (assignment: INormalizeAssignment) => void;
};

class AssignmentItem extends React.PureComponent<AssignmentItemProps> {
    handleSubmitAssignment = ({ assignmentRepo, studentComment }: AssignmentFormData) => {
        const { assignment } = this.props;
        const data = {
            ...assignment,
            assignment: {
                ...assignment.assignment,
                completeDate: Date.now(),
                assignmentRepo: assignmentRepo,
                studentComment: studentComment,
            },
        };
        this.props.updateAssignment(data);
    };

    render() {
        const { isEndAssignment } = this.props.assignment;
        const { status, title, urlToDescription, score, _id } = this.props.assignment.assignment;
        const card = setCardStyle(isEndAssignment, score, status);

        return (
            <Card className={`bm-3 mb-4 flex-grow-0 flex-shrink-1 ${cn('card')}  ${card.className}`}>
                <CardHeader className={`${card.cardHeader.className}`}>
                    {card.cardHeader.title}
                    {score !== 0 ? <span className={cn('score')}>{score}%</span> : null}
                </CardHeader>
                <AssignmentItemBody title={title} urlToDescription={urlToDescription} />
                {status === 'Assigned' &&
                    !isEndAssignment && (
                        <AssignmentItemForm onSubmit={this.handleSubmitAssignment} form={'form-' + _id.toString()} />
                    )}
            </Card>
        );
    }
}

export default AssignmentItem;
