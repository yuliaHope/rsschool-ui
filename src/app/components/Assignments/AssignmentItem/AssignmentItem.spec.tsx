import * as React from 'react';
import { shallow } from 'enzyme';
import AssignmentItem from './AssignmentItem';
import { AssignmentStatus } from '../../../core/models';

describe('Assignments Item', () => {
    const props = {
        assignment: {
            isEndAssignment: false,
            isActiveAssignment: true,
            assignment: {
                _id: 123,
                assignmentRepo: 'assignmentRepo',
                checkDate: 1555555555555555555555,
                completeDate: 10000555555555555,
                courseId: 'courseId',
                deadlineDate: 15555555555555555555555,
                mentorComment: 'mentorComment',
                mentorId: 'mentorId',
                score: 0,
                status: AssignmentStatus.Assigned,
                studentComment: 'studentComment',
                studentId: 'studentId',
                taskId: 'taskId',
                title: 'title',
                urlToDescription: 'urlToDescription',
                whoChecks: 'whoChecks',
            },
        },
        updateAssignment: jest.fn(),
    };

    describe('render card', () => {
        it('Assigned status', () => {
            const output = shallow(<AssignmentItem {...props} />);
            expect(output).toMatchSnapshot();
        });

        it('Checked status && score !== 100', () => {
            const nextProps = {
                ...props,
                assignment: {
                    ...props.assignment,
                    assignment: {
                        ...props.assignment.assignment,
                        status: AssignmentStatus.Checked,
                        score: 30,
                    },
                },
            };
            const output = shallow(<AssignmentItem {...nextProps} />);
            expect(output).toMatchSnapshot();
        });

        it('Checked status && score === 100', () => {
            const nextProps = {
                ...props,
                assignment: {
                    ...props.assignment,
                    assignment: {
                        ...props.assignment.assignment,
                        status: AssignmentStatus.Checked,
                        score: 100,
                    },
                },
            };
            const output = shallow(<AssignmentItem {...nextProps} />);
            expect(output).toMatchSnapshot();
        });

        it('Missed deadline', () => {
            const nextProps = {
                ...props,
                assignment: {
                    ...props.assignment,
                    isEndAssignment: true,
                },
            };
            const output = shallow(<AssignmentItem {...nextProps} />);
            expect(output).toMatchSnapshot();
        });
    });
});
