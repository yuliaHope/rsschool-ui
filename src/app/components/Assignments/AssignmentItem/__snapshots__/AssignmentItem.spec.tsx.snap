// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignments Item render card Assigned status 1`] = `
<Card
  className="bm-3 mb-4 flex-grow-0 flex-shrink-1 card  bg-secondary"
  tag="div"
>
  <CardHeader
    className=""
    tag="div"
  >
    Not submitted yet!
  </CardHeader>
  <AssignmentItemBody
    title="title"
    urlToDescription="urlToDescription"
  />
  <ReduxForm
    form="form-123"
    onSubmit={[Function]}
  />
</Card>
`;

exports[`Assignments Item render card Checked status && score !== 100 1`] = `
<Card
  className="bm-3 mb-4 flex-grow-0 flex-shrink-1 card  border-warning"
  tag="div"
>
  <CardHeader
    className="text-white bg-warning"
    tag="div"
  >
    Done!
    <span
      className="score"
    >
      30
      %
    </span>
  </CardHeader>
  <AssignmentItemBody
    title="title"
    urlToDescription="urlToDescription"
  />
</Card>
`;

exports[`Assignments Item render card Checked status && score === 100 1`] = `
<Card
  className="bm-3 mb-4 flex-grow-0 flex-shrink-1 card  border-success"
  tag="div"
>
  <CardHeader
    className="text-white bg-success"
    tag="div"
  >
    Done!
    <span
      className="score"
    >
      100
      %
    </span>
  </CardHeader>
  <AssignmentItemBody
    title="title"
    urlToDescription="urlToDescription"
  />
</Card>
`;

exports[`Assignments Item render card Missed deadline 1`] = `
<Card
  className="bm-3 mb-4 flex-grow-0 flex-shrink-1 card  border-danger"
  tag="div"
>
  <CardHeader
    className="text-white bg-danger"
    tag="div"
  >
    The deadline has passed!
  </CardHeader>
  <AssignmentItemBody
    title="title"
    urlToDescription="urlToDescription"
  />
</Card>
`;
